tests/integration/src/disaster_recovery_tests.rs

```diff
@@ -422,6 +418,85 @@ fn test_disaster_recovery_flow() {
     assert!(dr_status.recovery_requests.is_empty());
 }
 
+// The goal of this test is not to check if the committee is set correctly, but to check that when the station is
+// reinstalled, the upgrader canister is not recreated.
+#[test]
+fn test_disaster_recovery_flow_reuses_same_upgrader() {
+    // 1. setup the environment with one admin user
+    let TestEnv {
+        env, canister_ids, ..
+    } = setup_new_env();
+
+    let station_wasm_module = get_canister_wasm("station");
+    let system_info = get_system_info(&env, WALLET_ADMIN_USER, canister_ids.station);
+    let upgrader_id = system_info.upgrader_id;
+
+    // 2. perform the disaster recovery request with the station wasm and using the same upgrader id
+    let res: (ApiResult<()>,) = update_candid_as(
+        &env,
+        upgrader_id,
+        WALLET_ADMIN_USER,
+        "request_disaster_recovery",
+        (upgrader_api::RequestDisasterRecoveryInput {
+            module: station_wasm_module.clone(),
+            arg: Encode!(&station_api::SystemInstall::Init(station_api::SystemInit {
+                name: "Station".to_string(),
+                admins: vec![station_api::AdminInitInput {
+                    identity: WALLET_ADMIN_USER.clone(),
+                    name: "updated-admin-name".to_string(),
+                }],
+                quorum: None,
+                fallback_controller: None,
```

Comment:
Maybe we could provide one here to check against the bug mentioned above.

Reply:
The existing behaviour is to clear up the controllers in the init hook and make sure that only if a `fallback_controller` is specified that it would be set, so the test would be adding it where you suggested and then test that it is not there anymore, wdyt?

Reply:
> test that it is not there anymore

If there's a new fallback controller specified, then the station (and the upgrader) should have that new fallback controller eventually. The old fallback controller (if any) should ideally be preserved until the new fallback controller is set (the controller change should happen atomically).